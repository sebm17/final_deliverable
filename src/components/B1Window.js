import React, { useCallback } from "react";
import { fetchBSmart, fetchM98, generateB1File } from "../data";

function printInfo(data) {
  const entries = Object.entries(data).reduce((acc, [title, value]) => {
    if (value) {
      acc.push(
        <div>
          {title}: {value}
        </div>
      );
    }
    return acc;
  }, []);

  if (entries.length === 0) {
    return <div>No data to display</div>;
  }

  return entries;
}

export const B1Window = ({ selectedInput }) => {
  if (!selectedInput) {
    return null;
  }

  const bSmartData = fetchBSmart(selectedInput);
  const m98Data = fetchM98(selectedInput);

  const downloadURL = URL.createObjectURL(
    generateB1File({
      ...selectedInput,
      ...bSmartData,
      ...m98Data,
      elecSerialNumber: bSmartData.serialNumber,
      gasSerialNumber: m98Data.serialNumber
    })
  );

  return (
    <div className="b1Window three column row">
      <div className="column">
        <h1>Electricity</h1>
        {printInfo(m98Data)}
      </div>
      <div className="column">
        <h1>Gas</h1>
        {printInfo(bSmartData)}
      </div>
      <div className="column">
        <a href={downloadURL} download={"B1.csv"}>
          <button className="downloadButton ui button green">Download</button>
        </a>
      </div>
    </div>
  );
};
