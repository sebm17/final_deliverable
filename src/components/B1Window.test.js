import React from "react";
import { render } from "@testing-library/react";
import { B1Window } from "./B1Window";

test("if there is no input, do not render anything", async () => {
  const { queryByText } = render(<B1Window selectedInput={undefined} />);
  expect(queryByText("Gas")).toBeNull();
  expect(queryByText("Electricity")).toBeNull();
});

test("if there is input, render as expected", () => {
  // createObjectURL does not exist in jest so we pass a fake one to get the test passing
  global.URL = { ...global.URL, createObjectURL: jest.fn() };

  const wrapper = render(
    <B1Window
      selectedInput={{
        id: "1",
        mpan: "1591029068761",
        mprn: "7471856205",
        jad: "jad1",
        jtc: "jtc1"
      }}
    />
  );

  expect(wrapper).toMatchSnapshot();
});
