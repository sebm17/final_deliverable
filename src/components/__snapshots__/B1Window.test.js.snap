// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if there is input, render as expected 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="b1Window three column row"
      >
        <div
          class="column"
        >
          <h1>
            Electricity
          </h1>
          <div>
            appointmentId
            : 
            1331183424
          </div>
          <div>
            roleCode
            : 
            M
          </div>
          <div>
            serialNumber
            : 
            17K0068052
          </div>
          <div>
            meterOperatingMode
            : 
            S1
          </div>
        </div>
        <div
          class="column"
        >
          <h1>
            Gas
          </h1>
          <div>
            serialNumber
            : 
            G4K00122521706
          </div>
          <div>
            modelCode
            : 
            BK-G4E
          </div>
          <div>
            manufacturerCode
            : 
            ELM
          </div>
          <div>
            meterType
            : 
            S
          </div>
          <div>
            paymentMethod
            : 
            CR
          </div>
        </div>
        <div
          class="column"
        >
          <a
            download="B1.csv"
          >
            <button
              class="downloadButton ui button green"
            >
              Download
            </button>
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="b1Window three column row"
    >
      <div
        class="column"
      >
        <h1>
          Electricity
        </h1>
        <div>
          appointmentId
          : 
          1331183424
        </div>
        <div>
          roleCode
          : 
          M
        </div>
        <div>
          serialNumber
          : 
          17K0068052
        </div>
        <div>
          meterOperatingMode
          : 
          S1
        </div>
      </div>
      <div
        class="column"
      >
        <h1>
          Gas
        </h1>
        <div>
          serialNumber
          : 
          G4K00122521706
        </div>
        <div>
          modelCode
          : 
          BK-G4E
        </div>
        <div>
          manufacturerCode
          : 
          ELM
        </div>
        <div>
          meterType
          : 
          S
        </div>
        <div>
          paymentMethod
          : 
          CR
        </div>
      </div>
      <div
        class="column"
      >
        <a
          download="B1.csv"
        >
          <button
            class="downloadButton ui button green"
          >
            Download
          </button>
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
